tests/e2e/tkgs_ha_utils.go

```diff
@@ -0,0 +1,166 @@
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+// checkAnnotationOnSvcPvc checks tkg HA specific annotations on SVC PVC
+func checkAnnotationOnSvcPvc(svcPVC *v1.PersistentVolumeClaim,
+	allowedTopologies map[string][]string, categories []string) error {
+	framework.Logf("svcPVC: %v", svcPVC)
+	annotationsMap := svcPVC.Annotations
+	framework.Logf("annotationMap: %v", annotationsMap)
+	framework.Logf("accessible zone: %v, requested zones: %v", annotationsMap[tkgHAccessibleAnnotationKey], annotationsMap[tkgHARequestedAnnotationKey])
+	if accessibleTopoString, x := annotationsMap[tkgHAccessibleAnnotationKey]; x {
+		framework.Logf("accessible array: %T", accessibleTopoString)
+		accessibleTopology := strings.Split(accessibleTopoString, ":")
+		topoKey := strings.Split(accessibleTopology[0], "{")[1]
+		topoVal := strings.Split(accessibleTopology[1], "}")[0]
+		framework.Logf("Topo key: %v and val: %v", topoKey, topoVal)
+		category := strings.SplitAfter(topoKey, "/")[1]
+		categoryKey := strings.Split(category, `"`)[0]
+		if checkElePresentInList(categories, categoryKey) {
+			if checkElePresentInList(allowedTopologies[topoKey], topoVal) {
+				return fmt.Errorf("couldn't find allowed accessible topology: %v on svc pvc: %s"+
+					"instead found: %v", allowedTopologies[topoKey], svcPVC.Name, topoVal)
+			}
+		} else {
+			return fmt.Errorf("couldn't find key: %s on allowed categories %v",
+				category, categories)
+		}
+	} else {
+		return fmt.Errorf("couldn't find annotation key: %s on svc pvc: %s",
+			tkgHAccessibleAnnotationKey, svcPVC.Name)
+	}
+
+	if requestedTopoString, y := annotationsMap[tkgHARequestedAnnotationKey]; y {
+		framework.Logf("requested array: %T", requestedTopoString)
+		availabilityTopo := strings.Split(requestedTopoString, ",")
+		for _, avlTopo := range availabilityTopo {
+			requestedTopology := strings.Split(avlTopo, ":")
+			topoKey := strings.Split(requestedTopology[0], "{")[1]
+			topoVal := strings.Split(requestedTopology[1], "}")[0]
+			framework.Logf("Topo key: %v and val: %v", topoKey, topoVal)
+			category := strings.SplitAfter(topoKey, "/")[1]
+			categoryKey := strings.Split(category, `"`)[0]
+			if checkElePresentInList(categories, categoryKey) {
+				if checkElePresentInList(allowedTopologies[topoKey], topoVal) {
+					return fmt.Errorf("couldn't find allowed accessible topology: %v on svc pvc: %s"+
+						"instead found: %v", allowedTopologies[topoKey], svcPVC.Name, topoVal)
+				}
+			} else {
+				return fmt.Errorf("couldn't find key: %s on allowed categories %v",
+					category, categories)
+			}
+		}
+	} else {
+		return fmt.Errorf("couldn't find annotation key: %s on svc pvc: %s",
+			tkgHARequestedAnnotationKey, svcPVC.Name)
+	}
+	return nil
+}
+
+// checkElePresentInList is a util method which checks whether a particular string
+// is present in a given list or not
+func checkElePresentInList(strArr []string, str string) bool {
```

Comment:
"isValuePresentInTheList " Looks this name suits better . Think and take a call.